### PR TITLE
ft(backend):  Enable comments on requests

### PR DESCRIPTION
**Pull request by prieste**
**What does this PR do?**
   Allows user to comment on trip requests by carrying out CRUD operations on comments model

  **Description of Task to be completed?**
  - Creates a comments model
  - Creates routes for creating, getting and updating comments to requests
  - Updates swagger documentation
  -  Creates additional tests in test suite

  **How should this be manually tested?**
  - Fetch and checkout into this branch.
  - Run 'npm I' on your terminal.
  - Startup the server.
  -  navigate to postman.
  -  Sign up and create a request(using the request id as requestId in the route below) 
  -  Send a post request on the following route 'localhost:3000/api/v1/requests/{requestId}comments', providing a comments field on the body.
  - Send a get request on the same route above.
  -  Send a delete request on the following route  'localhost:3000/api/v1/requests/{requestId}comments','localhost:3000/api/v1/requests/{requestId}comments/{commentId}', where commentId is the Id of the comment create earlier.

  **Any background context you want to provide?**

  

  **What are the relevant stories?**

  #### pivotal tracker stories
 - [create](https://www.pivotaltracker.com/n/projects/2382051/stories/167735051)
 - [delete](https://www.pivotaltracker.com/n/projects/2382051/stories/167734993)

  **Screenshots (if appropriate)**

  
Screenshot from 2019-09-06 00-23-31
Screenshot from 2019-09-06 00-23-36

  **Questions:**

 N/A

### CODE REVIEW

File: src/controllers/commentController.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      badRequest(res, error);
+    }
+  }
+
+  /**
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} user - return object containing status and data
+ */
+  static async getComment({ params: { requestId } }, res) {
```
Comment (SamuelKoroh) [inline src/controllers/commentController.js]: You can just pass the req object to the getComment then destructure it from the commentService getComment method.

File: src/controllers/commentController.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      badRequest(res, error);
+    }
+  }
+
+  /**
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} user - return object containing status and data
+ */
+  static async deleteComment({ params: { commentId, requestId }, user: { id } }, res) {
```
Comment (SamuelKoroh) [inline src/controllers/commentController.js]: You can just pass the req object to the getComment then destructure it from the commentService getComment method.

File: src/controllers/commentController.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * Controller form request comments
+ */
+export default class CommentController {
+/**
+ * @param {object} res - response object
+ * @return {object} user - return object containing status and data
+ */
+  static async postComment({ body: { comment }, params: { requestId }, user: { id } }, res) {
+    try {
+      const data = await createComment(comment, requestId, id);
```
Comment (SamuelKoroh) [inline src/controllers/commentController.js]: You can just pass the req object to the getComment then destructure it from the commentService getComment method.

File: src/controllers/commentController.js (hunk 1 above)
Comment (prieste) [reply src/controllers/commentController.js]: Is there a reason you want me to do that? would you prefer it that way @Annmary12 ?

Comment (Annmary12) [reply src/controllers/commentController.js]: I will prefer to leave it as `req` and destructure inside the function

Comment (prieste) [reply src/controllers/commentController.js]: I've implemented feedback

**Comment by Annmary12** [comment]
@prieste , resolve conflict on your PR

File: src/controllers/commentController.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import Response from '../utils/response';
+import CommentService from '../services/commentService';
+
+const { createComment, getComment, deleteComment } = CommentService;
+const { success, badRequest } = Response;
+/**
```
Comment (Annmary12) [inline src/controllers/commentController.js:7]: Leave one line before the Jsdocs

File: src/controllers/commentController.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * Controller form request comments
+ */
+export default class CommentController {
+/**
```
Comment (Annmary12) [inline src/controllers/commentController.js:11]: Leave one line before the Jsdocs

File: src/tests/comments.test.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+  it('should get requests comments', async () => {
+    const { body: { data: [first] }, status } = await chai.request(app)
+      .get(`${URL_PREFIX}/${requestId}/comments`)
+      .set('token', authToken)
+      .send();
+    const { comment } = first;
+
+    expect(status).to.equal(200);
+    expect(comment).to.equal('this is a coment');
```
Comment (Annmary12) [inline src/tests/comments.test.js]: comment is misspelt as `coment`

File: src/tests/comments.test.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+  });
+
+  it('should throw an erro for invalid request id', async () => {
```
Comment (Annmary12) [inline src/tests/comments.test.js]: `erro` is misspelt

File: src/tests/comments.test.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+  });
+
+  it('should return an error when attempting to delete comment that does not exist', async () => {
+    const { body: { error }, status } = await chai.request(app)
+      .delete(`${URL_PREFIX}/${requestId}/comments/666`)
+      .set('token', authToken)
+      .send();
+
+    expect(status).to.equal(400);
+    expect(error).to.equal('Comment does not exist');
```
Comment (Annmary12) [inline src/tests/comments.test.js:215]: What happens when a user that is not the owner of a comment try to delete someone else comment? I don't think you implemented that

Comment (prieste) [reply src/tests/comments.test.js:215]: I use the users id (which I get from the token verifying middleware)as a foreign key on the comments table  so a user with a different id cannot delete a comment

**Pull request merged**